types/ramda/index.d.ts

```diff
@@ -281,7 +281,7 @@ export function comparator<T>(pred: (a: T, b: T) => boolean): (x: T, y: T) => nu
  * - applying g to zero or more arguments will give false if applying the same arguments to f gives
  *   a logical true value.
  */
-export function complement(pred: (...args: readonly any[]) => boolean): (...args: readonly any[]) => boolean;
+export function complement<As extends any[]>(pred: (...args: As) => boolean): (...args: As) => boolean;
```

Comment:
How about simplify as:
```suggestion
export function complement<F extends Pred>(pred: F): F;
```

Reply:
Because in cases with `value is [type]` like this:
```typescript
const isFoo = (value: string): value is "foo" => {
  return value === "foo"
}

// $ExpectType (value: string) => value is "foo"
const isNotFoo = complement(isFoo);
```
This behavior is incorrect

Reply:
Agreed, in such case returning `boolean` should be more appropriate semantically. Would you give test case to have it covered maybe?

Reply:
Test case added 👍